platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -78,9 +79,28 @@ public abstract class InitBuild extends DefaultTask {
     private ProjectLayoutSetupRegistry projectLayoutRegistry;
 
     /**
-     * The desired type of project to generate, defaults to 'pom' if a 'pom.xml' is found in the project root and if no 'pom.xml' is found, it defaults to 'basic'.
+     * Disables interactive dialog, ensuring no user input is required to complete the command.
+     * <p>
+     * This property can be set via the command-line options '--no-dialog' and '--dialog'.
      *
+     * @since 8.6
+     */
+    @Incubating
+    @Input
+    @Optional
+    @Option(option = "no-dialog", description = "Disable interactive dialog, use default values when none explicitly specified")
+    public abstract Property<Boolean> getNoDialog();
```

Comment:
I thought about it, and intentionally went for the "negative option", so that the description would be provided for the option that alters the default behavior of `init`, which is to have a dialog.

Otherwise, it would look like this:
```
Options
     --dialog     Enable interactive questions for configuring desired project
     --no-dialog     Disables option --dialog
```

Do you think it's still cleaner?

Reply:
This is a little odd to read: `Disables option --dialog`. 
Can we write something like `Disables dialog`, or `Disables --dialog`, or `Disables interactive dialog`?

Reply:
🤔 What about introducing a `no_description` option to `@Option`?

Reply:
🤔 Another idea, instead of making this focused on how it behaves (no interactive dialog), focus the name on what it means (accept defaults).

Instead of:
`gradle init --no-dialog --type kotlin-library`
You'd have:
`gradle init --accept-defaults --type kotlin-library`

The description would be something like "Generates build with default values where options are not explicitly specified"

Reply:
I like the idea of being able to provide a custom description for a `--no-` option. Though, I think that should be done separately. I created an issue:
https://github.com/gradle/gradle/issues/27391

Reply:
I also like the idea of giving the option the name that answers the "what?". Though, I think `--accept-defaults` is cumbersome to type.
What about:
- `--use-defaults`
- `--with-defaults`
- or even `--defaults`
?

I think we can sacrifice the readability of the opposite option that would start with `--no-`, because users would never have to type it.

Reply:
I like `--use-defaults` because it makes me think that in:`gradle init --use-defaults --type kotlin-library`, I want to init with all the defaults except for the type which I am overriding to a Kotlin library. It flows from a user perspective.

Reply:
I updated the PR to have `--use-defaults`